### ISSUE TITLE
Radio group

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#291]
## Environment
* **react-advanaced-form:** 1.5.0

## What
I propose to refine the concept of field (radio) group.

## Why
Grouping the fields may have a bigger value acting like a single control point (i.e. a group of N fields with the error in any of them being displayed only under the field group).

## How
This can be approached in multiple ways.

### Extending `Field.Group`
* Make `name` prop optional in `Field.Group` component
* Add a `render` prop for a `Field.Group` component. Render function accepts `fieldProps` and `fieldState`, allowing to render anything within the group based on the fields state.

> Caveats: How to manage `fieldState` for multiple fields?

### Add new component
...

**Comment by kettanaito** [comment]
`Field.Group` may have a custom `render` prop that exposes the `state` *array* containing each `fieldState` of the children fields. This way multiple states are iterable, and thus you can use them to have a custom logic for multiple fields.

```jsx
<Field.Group name="foo" render={(state) => (
  <Input name="firstName" />
  <Input name="lastName" />
  {state.some(fieldState => fieldState.invalid) && (
    <ErrorMessage>{state.filter(fieldState => fieldState.invalid).map(fieldState => (
      <span>fieldState.error</span>
    ))}</ErrorMessage>
  )}
)} />
```

**Comment by kettanaito** [comment]
I have merged a good example of a compound field with a single control point: [`BirthDate`](https://github.com/kettanaito/react-advanced-form/blob/master/examples/fields/BirthDate.jsx). Implemented using the current API (`1.5.0`).

**Comment by kettanaito** [comment]
After the proof of concept implementation for a complex field (BirthDate), I see no reason of adding a new API. `createField` is powerful enough to allow developers to maintain their own record in the field's state, reference and update them. There is no problem to group N amount of fields under a single field, as long as you control it the way you want. Validation, serialization, and all the other form events come built-in with `createField`.

Closing.
